src/tools/perf/perftest.c

```diff
@@ -782,18 +801,30 @@ static ucs_status_t read_batch_file(FILE *batch_file, const char *file_name,
 
 static ucs_status_t parse_cpus(char *opt_arg, struct perftest_context *ctx)
 {
-    char *token;
+    char *endptr, *cpu_list = opt_arg;
+    int cpu;
 
     ctx->num_cpus = 0;
-    token         = strtok(opt_arg, ",");
+    cpu           = strtol(cpu_list, &endptr, 10);
+
+    while (((*endptr == ',') || (*endptr == '\0')) && (ctx->num_cpus < MAX_CPUS)) {
+        if (cpu < 0) {
+            ucs_error("invalid cpu number detected: (%d)", cpu);
+            return UCS_ERR_INVALID_PARAM;
+        }
+
+        ctx->cpus[ctx->num_cpus++] = cpu;
+
+        if (*endptr == '\0') {
+            break;
+        }
 
-    while((token != NULL) && (ctx->num_cpus < MAX_CPUS)) {
-        ctx->cpus[ctx->num_cpus++] = atoi(token);
-        token = strtok(NULL, ",");
+        cpu_list                   = endptr + 1; /* skip the comma */
+        cpu                        = strtol(cpu_list, &endptr, 10);
```

Comment:
can use less spaces here (it does not have to be aligned with line 816 because there is if in between)